platform/commonUI/edit/src/actions/SaveAction.js

```diff
@@ -34,11 +34,13 @@ define(
         function SaveAction(
             dialogService,
             notificationService,
+            editNotesService,
```

Comment:
It would be nice to avoid taking on this dependency for the SaveAction (and other actions). Any mutations to domain objects should get persisted here; I don't think there's any reason that we'd need to wait until the end to mutate.